arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -787,25 +789,29 @@ Result RocksDBCollection::truncateWithRangeDelete(transaction::Methods& trx) {
   if (!s.ok()) {
     return rocksutils::convertStatus(s);
   }
+  // we need crash server if any failure in order to trigger recovery,
+  // as rocksdb truncate already committed here
```

Comment:
Because it is not explicitly mentioned that the way to crash the server is to mark the function as `noexcept`. One has to reverse-engineer the motivation from the code, not from the comment.
A clearer comment would have been something like
"We intentionally mark this lambda noexcept, so that when any exception occurs inside the lambda (can happen in case of out of memory), we crash the server instead of continuing with an inconsistent state. The post-crash recovery will then get everything into a consistent state again.".